sherpa-onnx/csrc/online-punctuation-cnn-bilstm-impl.h

```diff
@@ -0,0 +1,262 @@
+ public:
+  explicit OnlinePunctuationCNNBiLSTMImpl(
+      const OnlinePunctuationConfig &config)
+      : config_(config), model_(config.model) {
+        if (!config_.model.bpe_vocab.empty()) {
+          bpe_encoder_ = std::make_unique<ssentencepiece::Ssentencepiece>(
+              config_.model.bpe_vocab);
+        }
+      }
+
+#if __ANDROID_API__ >= 9
+  OnlinePunctuationCNNBiLSTMImpl(AAssetManager *mgr,
+                                      const OnlinePunctuationConfig &config)
+      : config_(config), model_(mgr, config.model) {
+        if (!config_.model.bpe_vocab.empty()) {
+          auto buf = ReadFile(mgr, config_.model.bpe_vocab);
+          std::istringstream iss(std::string(buf.begin(), buf.end()));
+          bpe_encoder_ = std::make_unique<ssentencepiece::Ssentencepiece>(iss);
+        }
+      }
+#endif
+
+  void encode_sentences(const std::string& text,
+                        std::vector<int32_t>& tokens_list,
+                        std::vector<int32_t>& valids_list,
+                        std::vector<int32_t>& label_len_list) const {
+    std::vector<int32_t> tokens;
+    std::vector<int32_t> valids;
+    int32_t label_len = 0;
+
+    tokens.push_back(1); // hardcode 1 now, 1 - <s>
+    valids.push_back(1);
+
+    std::stringstream ss(text);
+    std::string word;
+    while (ss >> word) {
+        std::vector<int32_t> word_tokens;
+        bpe_encoder_->Encode(word, &word_tokens); 
+
+        if (tokens.size() + word_tokens.size() > MAX_SEQ_LEN - 1) {
+          tokens.push_back(2); // hardcode 2 now, 2 - </s>
+          valids.push_back(1);
+
+          label_len = std::count(valids.begin(), valids.end(), 1);
+
+          while (tokens.size() < MAX_SEQ_LEN) {
+            tokens.push_back(0);
+            valids.push_back(0);
+          }
+
+          assert(tokens.size() == MAX_SEQ_LEN);
+          assert(valids.size() == MAX_SEQ_LEN);
+
+          tokens_list.insert(tokens_list.end(), tokens.begin(), tokens.end());
+          valids_list.insert(valids_list.end(), valids.begin(), valids.end());
+          label_len_list.push_back(label_len);
+
+          std::vector<int32_t>().swap(tokens);
+          std::vector<int32_t>().swap(valids);
+          label_len = 0;
+          tokens.push_back(1); // hardcode 1 now, 1 - <s>
+          valids.push_back(1);
+        }
+
+        tokens.insert(tokens.end(), word_tokens.begin(), word_tokens.end());
+        valids.push_back(1); // only the first sub word is valid
+        int32_t remaining_size = word_tokens.size() - 1;
+        while (remaining_size > 0) {
+          valids.push_back(0);
+          remaining_size--;
+        }
+    }
+
+    if (tokens.size() > 0) {
+      tokens.push_back(2); // hardcode 2 now, 2 - </s>
+      valids.push_back(1);
+
+      label_len = std::count(valids.begin(), valids.end(), 1);
+
+      while (tokens.size() < MAX_SEQ_LEN) {
+        tokens.push_back(0);
+        valids.push_back(0);
+      }
```

Comment:
```suggestion
      while (tokens.size() < MAX_SEQ_LEN) {
        tokens.push_back(0);
        valids.push_back(0);
      }
```

Please replace them with `resize()`.